### PR TITLE
Fixed a few eslint warnings

### PR DESCRIPTION
**Pull request by Tobbe**
I've fixed 20 eslint warnings. (106 down to 86.)
I didn't look at all the `any` warnings. 

There are a couple of other warnings that I plain didn't know how to fix. Plz help? It's "Don't use `Function` as a type." in `mockRequests.ts` and "Don't use `{}` as a type" in `vscode.ts`. Both can totally be left to be solved in another PR.

As long as no one spots any error in what I've done I think this is good to merge, and then we can keep squashing those warnings in future PRs.

### CODE REVIEW

File: packages/forms/src/index.tsx [hunk 1]
```diff
@@ -95,8 +95,8 @@ const inputTagProps = <T extends InputTagProps>(
   const {
     errorClassName,
     errorStyle,
-    dataType,
-    transformValue,
+    dataType, // eslint-disable-line @typescript-eslint/no-unused-vars
+    transformValue, // eslint-disable-line @typescript-eslint/no-unused-vars
```
Comment (peterp) [inline packages/forms/src/index.tsx:99]: I think you can prefix these with underscore and they should no longer throw errors.

```suggestion
    _dataType,
    _transformValue,
```

**Review by peterp — APPROVED** [review]
This is great, thanks Tobbe

File: packages/forms/src/index.tsx (hunk 1 above)
Comment (Tobbe) [reply packages/forms/src/index.tsx:99]: You mean like this?

```ts
  const {
    errorClassName,
    errorStyle,
    dataType: _dataType,
    transformValue: _tranformValue,
    ...tagProps
  } = props
  ```

That still gives me the warnings

```terminal
C:\Users\tobbe\dev\redwood\redwood\packages\forms\src\index.tsx
   61:40  warning  Unexpected any. Specify a different type             @typescript-eslint/no-explicit-any
   98:15  warning  '_dataType' is assigned a value but never used       @typescript-eslint/no-unused-vars
   99:21  warning  '_tranformValue' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

Comment (peterp) [reply packages/forms/src/index.tsx:99]: Oh sorry, I hadn't read the code properly. Your method is fine.

**Pull request merged**